### PR TITLE
[Fixes] Fix Set-EnvironmentOnAgent param mapping

### LINKED ISSUE #3716: [Bug Report]: Set-EnvironmentOnModule not correctly using the parameter
### Describe the bug

The `Set-EnvironmentOnAgent` function has a parameter called `PSModules` which is never being used in the function itself. The current structure of the pipeline template/action using the script "luckily" is defining the `$Modules` variable (yes, its currently PascalCapsed).

If one would call the function with another variable set in the running process, it would not work as expected.

Kudos to @akata72 for discovering the issue.

### To reproduce

Search for `Set-EnvironmentOnAgent -` and update the variable used in the call, currently `$Modules`, to something else like `$ModuleList`. Branch, commit, push, run and see that the modules are not handled.

### Code snippet

```Bicep
function Set-EnvironmentOnAgent {

    [CmdletBinding()]
    param (
        [Parameter(Mandatory = $false)]
        [Hashtable[]] $PSModules = @()
    )

(...)

    ####################################
    ##   Install PowerShell Modules   ##
    ####################################

(...)

    $Modules | ForEach-Object {
        Write-Verbose ('- {0}. [{1}]' -f $count, $_.Name) -Verbose
        $count++
    }
```

### Relevant log output

```Shell
NA
```

### PR DESCRIPTION
**Pull request by MariusStorhaug**
# Description

Fixes #3716 

- `utilities\pipelines\sharedScripts\Set-EnvironmentOnAgent.ps1`
  - Added an additional example for usage of the parameter.
  - Changed internal references from `$modules` to `$PSModules`.
- yml files
  - Update `$Modules` variable name to camelCasing.

## Pipeline references

| Pipeline |
| - |
| NA |

Link to a working section, the rest of the pipeline breaks due to PE not being updated on the given branch.

https://github.com/MariusStorhaug/ResourceModules/actions/runs/5974422417/job/16208545446#step:3:108

# Type of Change

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: settings.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ variables:
   # ---------------------- #
 
   templateSpecsDoPublish: true # Set to true, if you would like to publish module templates as template specs
-  templateSpecsRGName: 'artifacts-rg' # The name of the resource group to publish to. If the resource group does not exist, it will be created.
+  templateSpecsRGName: 'templateSpecs-rg' # The name of the resource group to publish to. If the resource group does not exist, it will be created.
```
Comment (eriqua) [inline settings.yml:62]: 👻

File: settings.yml [hunk 2]
```diff
@@ -68,8 +68,8 @@ variables:
   # ------------------------------- #
 
   bicepRegistryDoPublish: true # Set to true, if you would like to publish module templates to a bicep registry
-  bicepRegistryName: adpsxxazacrx001 # The name of the bicep registry (ACR) to publish to. If it does not exist, it will be created.
-  bicepRegistryRGName: 'artifacts-rg' # The resource group that hosts the private bicep registry (ACR)
+  bicepRegistryName: 'msxcr' # The name of the bicep registry (ACR) to publish to. If it does not exist, it will be created.
```
Comment (eriqua) [inline settings.yml:71]: 👻

File: settings.yml [hunk 3]
```diff
@@ -68,8 +68,8 @@ variables:
...
+  bicepRegistryRGName: 'registry-rg' # The resource group that hosts the private bicep registry (ACR)
```
Comment (eriqua) [inline settings.yml:72]: 👻

File: settings.yml (hunk 1 above)
Comment (MariusStorhaug) [reply settings.yml]: Ghostbusters!

File: settings.yml (hunk 2 above)
Comment (MariusStorhaug) [reply settings.yml]: Ghostbusters!

File: settings.yml (hunk 3 above)
Comment (MariusStorhaug) [reply settings.yml]: Who you gonna call... ?

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**